narwhals/_pandas_like/utils.py

```diff
@@ -417,9 +417,16 @@ def reverse_translate_dtype(  # noqa: PLR0915
 
 
 def validate_indices(series: list[PandasLikeSeries]) -> list[Any]:
-    idx = series[0]._native_series.index
-    reindexed = [series[0]._native_series]
-    for s in series[1:]:
+    lengths = [len(s) for s in series]
+    max_length = max(lengths)
+
+    idx = series[lengths.index(max_length)]._native_series.index
```

Comment:
This is the first index that has max length, hence the left-most non scalar index will still guide the resulting index